src/Voron/Impl/Journal/WriteAheadJournal.cs

```diff
@@ -1606,7 +1606,12 @@ public void ReduceSizeOfCompressionBufferIfNeeded(bool forceReduce = false)
         {
             var initialSize = _env.Options.InitialFileSize ?? _env.Options.InitialLogFileSize;
             if (ShouldReduceSizeOfCompressionPager(initialSize, forceReduce) == false)
+            {
+                //PERF: Compression buffer will be reused, it is safe to discard the content to clear the modified bit.
+                _compressionPager.DiscardWholeFile();
```

Comment:
I think we _always_ want to do that, even if we are about to close the file, no?

Reply:
Also, we can do this _immediately_ after writing to the journal, why wait for the next round?

Reply:
@ayende If we discard this after each commit then next time we do the commit the OS will need to map the file again, won't it? Wouldn't it affect the overall commit performance?

Reply:
ok